### PR TITLE
BOOST-4494 creating type head input lib

### PR DESCRIPTION
**Pull request by sachith94-lion**
added the type head input component as lib.

@DevDHera  some configurations are remaining. cc @janithaaws @LahiruDraco

### CODE REVIEW

**Comment by DevDHera** [comment]
@sachith94-lion don't think we need to use internal jira refs for these projects, also try to use https://www.conventionalcommits.org/en/v1.0.0/

**Comment by sachith94-lion** [comment]
@DevDHera sure will do that in my next commit

File: package.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    "start": "ng serve",
+    "build": "ng build",
+    "build-lib": "ng build typeHeadInput",
+    "pack-lib": "cd dist/type-head-input && npm pack",
+    "watch": "ng build --watch --configuration development",
+    "test": "ng test"
+  },
+  "private": true,
+  "dependencies": {
+    "@angular/animations": "^15.1.4",
```
Comment (DevDHera) [inline package.json:15]: @sachith94-lion can't we move to latest versions?

File: projects/type-head-input/src/lib/pipe/formControl.pipe.ts
Comment (DevDHera) [inline projects/type-head-input/src/lib/pipe/formControl.pipe.ts:1]: @sachith94-lion use kebab case for files

**Review by DevDHera — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (sachith94-lion) [reply package.json:15]: @DevDHera Currently, I added what in my current project used is it ok to update into 17?

File: projects/type-head-input/src/lib/pipe/formControl.pipe.ts
Comment (sachith94-lion) [reply projects/type-head-input/src/lib/pipe/formControl.pipe.ts:1]: @DevDHera `form-control` ?

File: package.json (hunk 1 above)
Comment (DevDHera) [reply package.json:15]: @sachith94-lion please update to v17

File: projects/type-head-input/src/lib/pipe/formControl.pipe.ts
Comment (DevDHera) [reply projects/type-head-input/src/lib/pipe/formControl.pipe.ts:1]: @sachith94-lion correct

**Pull request closed without merging**